src/Service/AppSearch/AppSearchService.php

```diff
@@ -82,6 +82,19 @@ public function addDocuments(array $items): BatchDocumentInterface
             }
 
             $fields = $this->getBuilder()->toArray($item);
+
+            $documentMaxSize = $this->getConfiguration()->getDocumentMaxSize();
+            if ($documentMaxSize  && strlen(serialize($fields)) >= $documentMaxSize) {
+                trigger_error(sprintf(
+                    'Document title: %s, Document class: %s, Document ID %s: exceeds maximum allowed document size of %s bytes',
+                    $fields['title'],
+                    $fields['source_class'],
+                    $fields['record_id'],
+                    $documentMaxSize
+                ), E_USER_WARNING);
```

Comment:
Let's upgrade this to an `IndexServiceException`, because we'll need to throw the job into a failure state to get retries.

Reply:
Sweet i'll swap it out mate

Reply:
Using `IndexServiceException` causes the bulk reindex to break. If works fine for a single record but when bulk reindexing the exception stops the reindex.